py2many/rewriters.py

```diff
@@ -262,3 +263,28 @@ def visit_Call(self, node):
                     else:
                         return self._do_other_rewrite(node)
         return node
+
+
+class StrStrRewriter(ast.NodeTransformer):
+    def __init__(self, language):
+        super().__init__()
+        self._language = language
+
+    def visit_Compare(self, node):
+        if self._language in {"dart", "kotlin"}:
+            return node
+
+        if isinstance(node.ops[0], ast.In):
+            left = node.left
+            right = node.comparators[0]
+            left_type = get_id(get_inferred_type(left))
+            right_type = get_id(get_inferred_type(right))
+            if left_type == "str" and right_type == "str":
+                ret = ast.parse("a.find(b) != -1").body[0].value
```

Comment:
Before settling on one canonical way to achieve str-in-str, I'd like to ensure we can easily implement that in each language.  To be useful for https://github.com/adsharma/py2many/issues/258, it must work in all languages.

The reason I dont like this approach is that the `-1` on the rhs is a magic value defined by the function on the lhs.  Each language is going to need to detect `.find` and then replace `-1` with whatever their function returns.

IMO it is better for each language to handle `'a' in 'ba'` directly and implement it using the best "contains" routine that can produce a boolean.  We already have one `if self._language` in `py2many/`, and IMO we should not add more.

---

Re `str.find`, Nim returns -1.  I am surprised that the str2str test case should have reported an error on this PR, as this change here should be sufficient to cause nim to need to be added to the `EXPECTED_SUCCESSES`.

Go's equivalent is strings.Index, and uses -1.

For C++, str.find returns `string::npos`.  If that is somehow handled by py2many to be `-1`, it should work. 

Julia has https://docs.julialang.org/en/v1/base/strings/index.html#Base.findfirst-Tuple{AbstractString,%20AbstractString} and it looks like the value to expect is `nothing`.  https://docs.julialang.org/en/v1/base/strings/#Base.occursin would be easier to implement as it returns bool.

On Rust, str.find uses `Result`, and that doesn't work in py2many yet.  `str.contains` avoids that headache as it returns bool.

Reply:
Agreed and addressed in the updated commit.